File: solutions/python3/easy/collections-counter/solution-approach1.py
# ──────────────────────────────────────────────────
# Link        https://www.hackerrank.com/challenges/collections-counter/problem?isFullScreen=true
# Problem     collections.Counter()
# Difficulty  Easy
# Subdomain   Collections
# Platform    HackerRank
# Language    python3
# Status      Accepted
# Submitted   2026-07-17, 09:59 p.m.
# ──────────────────────────────────────────────────

# Enter your code here. Read input from STDIN. Print output to STDOUT
from collections import Counter

# Read total number of shoes
num_shoes = int(input())

# Read shoe sizes and convert them to a list of integers
shoe_sizes = list(map(int, input().split()))

# Create a inventory counter
inventory = Counter(shoe_sizes)

# Read total number of customers
num_customers = int(input())

total_earned = 0

# Process each customer's request
for _ in range(num_customers):
    size, price = map(int, input().split())
    
    # Check if the shoe size is available in stock
    if inventory[size] > 0:
        total_earned += price
        inventory[size] -= 1  # Reduce inventory by one shoe

# Print the final earnings
print(total_earned)

File: solutions/c/medium/printing-pattern-2/solution-approach1.c
// ──────────────────────────────────────────────────
// Link        https://www.hackerrank.com/challenges/printing-pattern-2/problem?isFullScreen=true
// Problem     Printing Pattern Using Loops
// Difficulty  Medium
// Subdomain   Conditionals and Loops
// Platform    HackerRank
// Language    c
// Status      Accepted
// Submitted   2026-08-21, 11:57 p.m.
// ──────────────────────────────────────────────────

#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

int main() 
{

    int n;
    scanf("%d", &n);
  	// Complete the code to print the pattern.
    int size = 2 * n - 1;   

    for (int i = 0; i < size; i++) {
        for (int j = 0; j < size; j++) {

            int top = i;
            int left = j;
            int bottom = size - 1 - i;
            int right = size - 1 - j;

            int dist = top;
            if (left < dist) dist = left;
            if (bottom < dist) dist = bottom;
            if (right < dist) dist = right;

            int value = n - dist;

            printf("%d", value);

            if (j != size - 1) {
                printf(" ");
            }
        }
        printf("\n");    
    }
    return 0;
}
